cnf-certification-test/accesscontrol/securitycontextcontainer/securitycontextcontainer.go

```diff
@@ -0,0 +1,240 @@
+	containerSCC.HostPorts = false
+	for _, aPort := range cut.Ports {
+		if aPort.HostPort != 0 {
+			containerSCC.HostPorts = true
+			break
+		}
+	}
+	containerSCC = updateCapabilities(cut, containerSCC)
+	if cut.SecurityContext != nil && cut.SecurityContext.AllowPrivilegeEscalation != nil {
+		containerSCC.PrivilegeEscalation = true
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.Privileged != nil {
+		containerSCC.PrivilegedContainer = *(cut.SecurityContext.Privileged)
+	}
+
+	if cut.SecurityContext != nil && cut.SecurityContext.RunAsUser != nil {
+		containerSCC.RunAsUser = true
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.ReadOnlyRootFilesystem != nil {
+		containerSCC.ReadOnlyRootFilesystem = *cut.SecurityContext.ReadOnlyRootFilesystem
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.RunAsNonRoot != nil {
+		containerSCC.RunAsNonRoot = *cut.SecurityContext.RunAsNonRoot
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.SELinuxOptions != nil {
+		containerSCC.SeLinuxContext = true
+	} else {
+		containerSCC.SeLinuxContext = false
+	}
+	return containerSCC
+}
+
+func updateCapabilities(cut *v1.Container, containerSCC ContainerSCC) ContainerSCC {
+	containerSCC.HaveDropCapabilities = false
+	if cut.SecurityContext != nil && cut.SecurityContext.Capabilities != nil {
+		var sliceDropCapabilities []string
+		for _, ncc := range cut.SecurityContext.Capabilities.Drop {
+			sliceDropCapabilities = append(sliceDropCapabilities, string(ncc))
+		}
+		sort.Strings(sliceDropCapabilities)
+		sort.Strings(requiredDropCapabilities)
+		containerSCC.HaveDropCapabilities = reflect.DeepEqual(sliceDropCapabilities, requiredDropCapabilities)
+		if checkContainCateegory(cut.SecurityContext.Capabilities.Add, category3AddCapabilities) {
+			containerSCC.Capabilities = "category3"
+		} else {
+			if checkContainCateegory(cut.SecurityContext.Capabilities.Add, category4AddCapabilities) {
+				containerSCC.Capabilities = "category4"
+			} else {
+				if len(cut.SecurityContext.Capabilities.Add) > 0 {
+					containerSCC.Capabilities = "category5"
+				} else {
+					containerSCC.Capabilities = ""
+				}
+			}
+		}
+	} else {
+		containerSCC.Capabilities = ""
+	}
+	return containerSCC
+}
+func contains(elems []string, v string) bool {
+	for _, s := range elems {
+		if v == s {
+			return true
+		}
+	}
+	return false
+}
+
+func AllVolumeAllowed(volumes []v1.Volume) bool {
+	countVolume := 0
+	for j := 0; j < len(volumes); j++ {
+		if volumes[j].ConfigMap != nil {
+			countVolume++
+		}
+		if volumes[j].DownwardAPI != nil {
+			countVolume++
+		}
+		if volumes[j].EmptyDir != nil {
+			countVolume++
+		}
+		if volumes[j].PersistentVolumeClaim != nil {
+			countVolume++
+		}
+		if volumes[j].Projected != nil {
+			countVolume++
+		}
+		if volumes[j].Secret != nil {
+			countVolume++
+		}
+	}
+	return countVolume == len(volumes)
+}
+
+func Checkcategory(containers []v1.Container, containerSCC ContainerSCC) []string {
```

Comment:
Adjust the func name for camel case.

```suggestion
func CheckCategory(containers []v1.Container, containerSCC ContainerSCC) []string {
```